torch/jit/_shape_functions.py

```diff
@@ -837,6 +837,22 @@ def flatten(input: List[int], start_dim: int, end_dim: int):
         shape.append(input[i])
     return shape
 
+def _reduce_along_dim(self: List[int], dim: int, keepdim: bool):
```

Comment:
So, the way that this works is that we compile these in torchscript then write the serialized functions out to a cpp file so the shape functions will work without a python dependency.. this process depends on torchscript being already built, so unfortunately we can automate this in the build step. 

You need to run `cd ~/pytorch/ && python torchgen/shape_functions/gen_jit_shape_functions.py` after making the changes here. 

Also, there was a recent moving of these files so it needs a small change in order to work:

```
diff --git a/torchgen/shape_functions/gen_jit_shape_functions.py b/torchgen/shape_functions/gen_jit_shape_functions.py
index cd43843600..d25539d3fa 100644
--- a/torchgen/shape_functions/gen_jit_shape_functions.py
+++ b/torchgen/shape_functions/gen_jit_shape_functions.py
@@ -114,7 +114,7 @@ def write_decomposition_util_file(path: str) -> None:
 
 
 def main() -> None:
-    pytorch_dir = Path(__file__).resolve().parents[3]
+    pytorch_dir = Path(__file__).resolve().parents[2]
     upgrader_path = pytorch_dir / "torch" / "csrc" / "jit" / "runtime"
     write_decomposition_util_file(str(upgrader_path))
```

Reply:
That script will not work until this diff is landed and you rebase onto it. 
https://github.com/pytorch/pytorch/pull/76703